### PR TITLE
Read to byte array method for client.AsyncResponse

### PR DESCRIPTION
**Pull request by er1c**

### CODE REVIEW

**Comment by er1c** [comment]
Aaaand apparently that only gets the first chunk converted :(

iterm

File: src/main/scala/fm/http/LinkedHttpContentReader.scala [hunk 1]
```diff
@@ -39,6 +39,20 @@ final class LinkedHttpContentReader(is100ContinueExpected: Boolean, head: Future
   private[this] val foldLeftCalled: AtomicBoolean = new AtomicBoolean(false)
   @volatile private[this] var current: Future[Option[LinkedHttpContent]] = head
 
+  // Access the LinkedHttpContent
+  def useLinkedHttpContent[T](f: LinkedHttpContent => Future[T]): Future[T] = {
```
Comment (tpunder) [inline src/main/scala/fm/http/LinkedHttpContentReader.scala]: I'm not sure what you think this accomplishes...  All you've really done is create an accessor for the `current` variable.  There's a reason everything goes through the foldLeft method.

File: src/main/scala/fm/http/LinkedHttpContentReader.scala [hunk 2]
```diff
@@ -39,6 +39,20 @@ final class LinkedHttpContentReader(is100ContinueExpected: Boolean, head: Future
...
+    if (is100ContinueExpected) ctx.writeAndFlush(CONTINUE)
+    
+    ctx.read()
+    
+    current.flatMap{ lc: Option[LinkedHttpContent] =>      
+      lc match {
+        case None => Future.failed(throw new Exception("No Content"))
```
Comment (tpunder) [inline src/main/scala/fm/http/LinkedHttpContentReader.scala]: This doesn't do what you think it does...

File: src/main/scala/fm/http/client/Response.scala [hunk 3]
```diff
@@ -110,7 +111,13 @@ final class AsyncResponse (response: HttpResponse, content: LinkedHttpContentRea
       true
     }
   }
-  
+
+  def toServerAsyncResponse(): Future[ServerAsyncResponse] = {
+    content.useLinkedHttpContent[ServerAsyncResponse] { content: LinkedHttpContent =>
+      Future.successful(ServerAsyncResponse(status, headers, content))
```
Comment (tpunder) [inline src/main/scala/fm/http/client/Response.scala]: What's the point of the `Future.successful(...)` ?

**Comment by er1c** [comment]
PR Updated, method to read to a byte array instead of a string

**Comment by er1c** [comment]
Added an httpClient.getFullBytes method that returns a FullBytesResponse that has body as the raw byte array instead of a string

**Pull request merged**